### PR TITLE
Fix: Move old features to the new main

### LINKED ISSUE #860: move old features to new main branch

### PR DESCRIPTION
**Pull request by shekhar2807**
# Description

- GetIteration concurrency
- Moved chainId to config
- Added compress parameter in logger.go
- Godoc documentation for utils, logger, core, metrics

Fixes #860

### CODE REVIEW

**Review by Yashk767 — DISMISSED** [review]
LGTM

**Review by ashish10677 — CHANGES_REQUESTED** [review]

File: logger/logger.go [hunk 1]
```diff
@@ -65,6 +67,7 @@ func InitializeLogger(fileName string) {
 			MaxSize:    logFileMaxSize,
 			MaxBackups: logFileMaxBackups,
 			MaxAge:     logFileMaxAge,
+			Compress:   true,
```
[the remark this answers is not in the document]
Comment (shekhar2807) [reply logger/logger.go]: Done @ashish10677

File: cmd/config-utils.go [hunk 2]
```diff
@@ -20,6 +20,7 @@ func (*UtilsStruct) GetConfigData() (types.Configurations, error) {
 		LogFileMaxSize:     0,
 		LogFileMaxBackups:  0,
 		LogFileMaxAge:      0,
+		Compress:           "",
```
Comment (ashish10677) [inline cmd/config-utils.go:23]: Is there any problem keeping this as a bool instead of a string?

Comment (shekhar2807) [reply cmd/config-utils.go:23]: Yes, I tried using that also . It was throwing some errors . As far as String is concerned I checked for every possibility. It works for every case.

Comment (shekhar2807) [reply cmd/config-utils.go:23]: Let me try that again, will update it here .

Comment (shekhar2807) [reply cmd/config-utils.go:23]: if compress != "" {
 		viper.Set("compress", compress)
} . This if statement cause the issue if compress is stated as boolean . String solves this issue and make it easier to implement. @ashish10677

**Review by ashish10677 — APPROVED** [review]
Lgtm

**Pull request merged**